core/storage/collection/gae_models.py

```diff
@@ -196,6 +196,35 @@ def get_deletion_policy():
         """
         return base_models.DELETION_POLICY.KEEP_IF_PUBLIC
 
+    @staticmethod
+    def transform_dict_to_valid(model_dict):
+        """Replace invalid fields and values in the CollectionRightsModel dict.
+
+        Some old CollectionRightsSnapshotContentModels can contain fields
+        and field values that are no longer supported and would cause
+        an exception when we would try to create CollectionRightsModel. We need
+        to remove or replace these fields and values.
+
+        Args:
+            model_dict: dict. The content of the model. Some fields and field
+                values might no longer exist in the CollectionRightsModel.
+
+        Returns:
+            dict. The content of the model. Only valid fields and values are
+            present.
+        """
+        # The status field could contain 'publicized' value in history, this
+        # value is now equivalent to 'public'.
+        if model_dict['status'] == 'publicized':
+            model_dict['status'] = constants.ACTIVITY_STATUS_PUBLIC
+        # The voice_artist_ids field was previously named translator_ids, we
```

Comment:
translator_ids, we --> translator_ids. We


Incidentally, one general thought: do you think the rights models should have schema migration methods, just like we do for the main Exploration model? That might be the "correct" way to handle this in the long term -- all versioned models should have schema upgrade functionality. I am a bit worried that, if future changes happen to these models, no one will remember to keep these functions up to date. What do you think?

Reply:
Although the `ExplorationModel` does have `states_schema_version` this is only used to track the version of the `states` field and it seems to me that the `schema_version` for the whole model is used only for reconstituting the `ExplorationModel` from the YAML files. I didn't find the usage of these migration methods anywhere else, for example when reverting.

Reply:
Oh yes, that's a good point, thanks for catching that. We may need a different general solution for handling primary fields in models, then.